controllers/wildflyserver_controller_test.go

```diff
@@ -580,3 +581,393 @@ func TestWildFlyServerWithSecurityContext(t *testing.T) {
 	assert.Equal(readOnlyRootFilesystem, statefulSet.Spec.Template.Spec.Containers[0].SecurityContext.ReadOnlyRootFilesystem)
 	assert.Equal(runAsNonRoot, statefulSet.Spec.Template.Spec.Containers[0].SecurityContext.RunAsNonRoot)
 }
+
+func TestWildFlyServerWithHttpProbes(t *testing.T) {
```

Comment:
@yersan I don't understand what triggers the usage of HTTP probes here; IIUC default (no `bootableJar` or `http` setting) should be exec probes

Reply:
@yersan shouldn't we expect exec probes like e.g. ```        exec:
          command:
            - /bin/bash
            - '-c'
            - >-
              if [ -f '/opt/eap/bin/livenessProbe.sh' ]; then
              /opt/eap/bin/livenessProbe.sh; else curl --fail
              http://127.0.0.1:9990/health/live; fi``` in this case?

Reply:
Hi @tommaso-borgato , the intention of this test is to verify the creation of probes that use pure HTTP request, not the Exec commands. I used the presence of the aforementioned environment variables to force the switch that triggers it. 

Would it get clearer if we use bootableJar=true? That would be more close to the reality.

The verification of the Exec probes is done in the TestWildFlyServerWithProbesScript

Reply:
> which hasn't either bootableJar or http set, when processed by the operator produces HTTP Probes instead of exec probes;

Because the env variables SERVER_LIVENESS_SCRIPT and SERVER_READINESS_SCRIPT are not configured for this test. If they are not configured (for example the upstream operator does not configure them), then we configure the probes using HTTP get requests.

Reply:
@yersan Thank you for the explanation, now I see it